src/fty_common_messagebus_unix_socket_helpers.cc

```diff
@@ -0,0 +1,185 @@
+                while (++frameIterator != payload.end())
+                {
+                    key = *frameIterator;
+
+                    if ( key == END_METADATA )
+                    {
+                        break;
+                    }
+
+                    ++frameIterator;
+                    const std::string & value = *frameIterator;
+
+                    //add to metaData
+                    message.metaData().emplace(key, value);
+                    printf("Metadata[%s]=%s\n",key.c_str(),value.c_str());
+                }
+            }
+            else
+            {
+                message.userData().emplace_back(key);
+                printf("Userdata <%s>\n",key.c_str());
+            }
+
+            while (++frameIterator != payload.end())
+            {
+                message.userData().emplace_back(*frameIterator);
+                printf("Userdata <%s>\n",(*frameIterator).c_str());
+            }
+        }
+
+        return message;
+    }
+    
+    void sendMessage(int socket, const Message & msg)
+    {
+        std::vector<std::string> payload;
+
+        //add meta data
+        payload.push_back(BEGIN_METADATA);
+        for (const auto& pair : msg.metaData())
+        {
+            payload.push_back(pair.first);
+            payload.push_back(pair.second);
+        }
+        payload.push_back(END_METADATA);
+
+        //add user data
+        for(const auto& item : msg.userData())
+        {
+            payload.push_back(item);
+        }
+
+        sendFrames(socket, payload);
+    }
+
+    std::vector<std::string> recvFrames(int socket)
+    {
+        //format => [ Number of frames ], [ <size of frame 1> <data> ], ... [ <size of frame N> <data> ]
+
+        //get the number of frames
+        uint32_t numberOfFrame = 0;
+
+        
+        if(read(socket, &numberOfFrame, sizeof(uint32_t)) != sizeof(uint32_t))
+        {
+            throw MessageBusException("Error while reading number of frame");
+        }
+        
+
+        //Get frames
+        std::vector<std::string> frames;
+
+        for( uint32_t index = 0; index < numberOfFrame; index++)
+        {
+            //get the size of the frame
+            uint32_t frameSize = 0;
+            
+            
+            if(read(socket, &frameSize, sizeof(uint32_t)) != sizeof(uint32_t))
+            {
+                throw MessageBusException("Error while reading size of frame");
+            }
+            
+
+            if(frameSize == 0)
+            {
+                throw MessageBusException("Read error: Empty frame");
+            }
+
+            //get the payload of the frame
+            std::vector<char> buffer;
+            buffer.reserve(frameSize+1);
+            buffer[frameSize] = 0;
+
+            if(read(socket, &buffer[0], frameSize) != frameSize)
+            {
+                throw MessageBusException("Read error while getting payload of frame");
+            }
+            
+            std::string str(&buffer[0]);
```

Comment:
This will break with binary data containing NULL characters. Use `std::string(const char *buf, size_t count)` constructor instead.